tests/pages/api/csvUpload.test.ts

```diff
@@ -376,6 +383,41 @@ describe('csvUpload', () => {
         expect(updateSessionAttributeSpy).toBeCalledWith(req, CSV_UPLOAD_ATTRIBUTE, { errors: mockError });
     });
 
+    it('should return 302 redirect to /matching if the fares triangle data has empty prices', async () => {
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: {},
+            body: null,
+            uuid: {},
+        });
+        const file = {
+            'csv-upload': {
+                size: 999,
+                path: 'string',
+                name: 'string',
+                type: 'text/csv',
+                toJSON(): string {
+                    return '';
+                },
+            },
+        };
+
+        getFormDataSpy.mockImplementation().mockResolvedValue({
+            files: file,
+            fileContents: csvData.missingPricesTestCsv,
+        });
+
+        jest.spyOn(fileUpload, 'containsViruses')
+            .mockImplementation()
+            .mockResolvedValue(false);
+
+        await csvUpload.default(req, res);
+
+        expect(res.writeHead).toBeCalledWith(302, {
+            Location: '/matching',
+        });
+        expect(updateSessionAttributeSpy).toBeCalledWith(req, CSV_UPLOAD_ATTRIBUTE, { errors: [] });
+    });
+
```

Comment:
Found a few things in testing with @matt-twomey  : 

- if we have one empty fare stage name, it is not currently picked up by the API - two or more is caught because they are duplicates of one another. We probably need to trim the stage names and see if they're empty. This is an old bug but we may as well fix it here.
- If the file uploaded has entirely empty prices, this is allowed by the system (and shouldn't be). We would also benefit from a unit test for this.
- Could we have a unit test to ensure that empty prices are not added to the data that gets compiled?